<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>EditDialog</class>
 <widget class="QDialog" name="EditDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>618</width>
    <height>382</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Edit database cell</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_5">
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QLabel" name="labelMode">
       <property name="minimumSize">
        <size>
         <width>0</width>
         <height>0</height>
        </size>
       </property>
       <property name="text">
        <string>Mode:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QComboBox" name="comboMode">
       <property name="minimumSize">
        <size>
         <width>0</width>
         <height>0</height>
        </size>
       </property>
       <item>
        <property name="text">
         <string>Text</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Binary</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Image</string>
        </property>
       </item>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="buttonImport">
       <property name="minimumSize">
        <size>
         <width>0</width>
         <height>0</height>
        </size>
       </property>
       <property name="toolTip">
        <string>Import text</string>
       </property>
       <property name="whatsThis">
        <string>Opens a file dialog used to import text to this database cell.</string>
       </property>
       <property name="text">
        <string>&amp;Import</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="buttonExport">
       <property name="minimumSize">
        <size>
         <width>0</width>
         <height>0</height>
        </size>
       </property>
       <property name="toolTip">
        <string>Export text</string>
       </property>
       <property name="whatsThis">
        <string>Opens a file dialog used to export the contents of this database cell to a text file.</string>
       </property>
       <property name="text">
        <string>&amp;Export</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="buttonNull">
       <property name="minimumSize">
        <size>
         <width>0</width>
         <height>0</height>
        </size>
       </property>
       <property name="toolTip">
        <string>Set this cell to NULL</string>
       </property>
       <property name="whatsThis">
        <string>Erases the contents of the cell</string>
       </property>
       <property name="text">
        <string>Set as &amp;NULL</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QStackedWidget" name="editorStack">
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="verticalLayout_4">
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <item>
        <widget class="QLabel" name="labelBinaryWarning">
         <property name="text">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600; color:#c00000;&quot;&gt;Warning: Editing binary content in text mode may result in corrupted data!&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QTextEdit" name="editorText">
         <property name="font">
          <font>
           <family>Monospace</family>
          </font>
         </property>
         <property name="whatsThis">
          <string>This area displays information about the data present in this database cell</string>
         </property>
         <property name="acceptRichText">
          <bool>false</bool>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="editorBinary"/>
     <widget class="QScrollArea" name="editorImageScrollArea">
      <property name="widgetResizable">
       <bool>true</bool>
      </property>
      <widget class="QWidget" name="scrollAreaWidgetContents">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>0</y>
         <width>83</width>
         <height>40</height>
        </rect>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <widget class="QLabel" name="editorImage">
          <property name="text">
           <string/>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_2">
     <item>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QLabel" name="labelType">
         <property name="text">
          <string>Type of data currently in cell</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="labelSize">
         <property name="text">
          <string>Size of data currently in table</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <spacer name="horizontalSpacer_2">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="buttonApply">
       <property name="text">
        <string>Apply</string>
       </property>
       <property name="default">
        <bool>true</bool>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>buttonExport</tabstop>
  <tabstop>buttonNull</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>buttonImport</sender>
   <signal>clicked()</signal>
   <receiver>EditDialog</receiver>
   <slot>importData()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>400</x>
     <y>28</y>
    </hint>
    <hint type="destinationlabel">
     <x>60</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonExport</sender>
   <signal>clicked()</signal>
   <receiver>EditDialog</receiver>
   <slot>exportData()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>488</x>
     <y>51</y>
    </hint>
    <hint type="destinationlabel">
     <x>162</x>
     <y>17</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>editorText</sender>
   <signal>textChanged()</signal>
   <receiver>EditDialog</receiver>
   <slot>editTextChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>234</x>
     <y>218</y>
    </hint>
    <hint type="destinationlabel">
     <x>362</x>
     <y>309</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>editorStack</sender>
   <signal>currentChanged(int)</signal>
   <receiver>comboMode</receiver>
   <slot>setCurrentIndex(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>185</x>
     <y>169</y>
    </hint>
    <hint type="destinationlabel">
     <x>149</x>
     <y>39</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonApply</sender>
   <signal>clicked()</signal>
   <receiver>EditDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>605</x>
     <y>358</y>
    </hint>
    <hint type="destinationlabel">
     <x>241</x>
     <y>406</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>comboMode</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>EditDialog</receiver>
   <slot>editModeChanged(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>71</x>
     <y>27</y>
    </hint>
    <hint type="destinationlabel">
     <x>201</x>
     <y>431</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonNull</sender>
   <signal>clicked()</signal>
   <receiver>EditDialog</receiver>
   <slot>setNull()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>528</x>
     <y>28</y>
    </hint>
    <hint type="destinationlabel">
     <x>723</x>
     <y>58</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>importData()</slot>
  <slot>exportData()</slot>
  <slot>editTextChanged()</slot>
  <slot>checkDataType()</slot>
  <slot>editModeChanged(int)</slot>
  <slot>setNull()</slot>
 </slots>
</ui>
